### PR TITLE
handler unit tests

### PR DESCRIPTION
**Pull request by barrytam20**
- changing database constructor to accept db connection params
- injecting database client into handler to aid testing
- handler unit tests

### CODE REVIEW

File: api/handler/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func Test_Handler_Ping(t *testing.T) {
```
Comment (mpuckett159) [inline api/handler/handler_test.go:11]: Is it normal convention for testing function to be snake case instead of camel?

**Review by mpuckett159 — COMMENTED** [review]
Pushing comments (questions) I have then going to run tests on another machine so just saving my work basically.

File: api/handler/handler.go [hunk 2]
```diff
@@ -7,15 +7,27 @@ import (
...
+type HandInterface interface {
+	Ping(w http.ResponseWriter, r *http.Request)
+	DescribeDepartments(w http.ResponseWriter, r *http.Request)
+	SeattleOfficerMetadata(w http.ResponseWriter, r *http.Request)
+	SeattleStrictMatch(w http.ResponseWriter, r *http.Request)
+	SeattleFuzzySearch(w http.ResponseWriter, r *http.Request)
+	TacomaOfficerMetadata(w http.ResponseWriter, r *http.Request)
+	TacomaStrictMatch(w http.ResponseWriter, r *http.Request)
+	TacomaFuzzySearch(w http.ResponseWriter, r *http.Request)
+}
```
[the remark this answers is not in the document]
Comment (barrytam20) [reply api/handler/handler.go:20]: good catch. not sure why i named it like that tbh

**Review by mpuckett159 — APPROVED** [review]
Looks great, 🎉 for having unit tests.

**Comment by barrytam20** [comment]
closing as changes are in #19

**Pull request closed without merging**